src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/WebClientConfiguration.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.hmppstemplatepackagename.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.netty.http.client.HttpClient
+import java.time.Duration
+import kotlin.apply as kotlinApply
+
+@Configuration
+class WebClientConfiguration(
+  @Value("\${api.base.url.oauth}") val oauthApiBaseUri: String,
+  @Value("\${api.health-timeout:2s}") val healthTimeout: Duration,
+  @Value("\${api.timeout:90s}") val timeout: Duration,
+) {
+  @Bean
+  fun oauthApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(oauthApiBaseUri, healthTimeout)
+
+  /**
+   * TODO
+   * Once you have a client registration defined in properties `spring.security.client.registration` then you'll
+   *  need to uncomment this @Bean and create both a health and authorized web client.
+   *
+   * e.g. if your client registration config looks like this (registrationId is `prison-api`):
+   * ```
+   * spring:
+   *   security:
+   *     client:
+   *       registration:
+   *            prison-api:
+   *             provider: hmpps-auth
+   *             client-id: ${prison-api.client.id}
+   *             client-secret: ${prison-api.client.secret}
+   *             authorization-grant-type: client_credentials
+   *             scope: read
+   * ```
+   * Then you need to create web clients in this class as follows:
+   * ```
+   *   @Bean
+   *   fun prisonApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(prisonApiBaseUri, healthTimeout)
+   *
+   *   @Bean
+   *   fun prisonApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager, builder: WebClient.Builder): WebClient =
+   *     builder.authorisedWebClient(authorizedClientManager, registrationId = "prison-api", url = prisonApiBaseUri, timeout)
+   * ```
+   */
+  // @Bean
+  fun authorizedClientManager(
+    clientRegistrationRepository: ClientRegistrationRepository,
+    oAuth2AuthorizedClientService: OAuth2AuthorizedClientService,
+  ): OAuth2AuthorizedClientManager {
```

Comment:
If you don't have any clients registrations then Spring doesn't create a ClientRegistrationRepository, hence the `@Bean` annotation is commented out.